### PR TITLE
VSCode workspace configuration

### PR DESCRIPTION
**Pull request by idkgene**
## Description

This PR optimizes VSCode settings by merging single language configuration into one, reducing code duplication.

## Related Issues

N/A

## Checklist

- [X] My code follows the code style of this project.
- [x] I have performed a self-review of my code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have updated the documentation, if necessary.
- [ ] I have added tests that prove my fix is effective or my feature works.
- [ ] New and existing tests pass locally with my changes.

## Additional Notes

Docs for this: https://code.visualstudio.com/docs/getstarted/settings#_multiple-languagespecific-editor-settings

### CODE REVIEW

**Comment by idkgene** [comment]
Hey @haydenbleasel, if you don't mind, I'll add extensions.json (from here - https://code.visualstudio.com/docs/editor/extension-marketplace#_workspace-recommended-extensions). We can tweak the list when I've prepared it

File: .vscode/settings.json [hunk 1]
```diff
@@ -8,20 +8,9 @@
...
-    "editor.defaultFormatter": "biomejs.biome"
-  },
-  "[javascript]": {
-    "editor.defaultFormatter": "biomejs.biome"
-  },
-  "[jsonc]": {
-    "editor.defaultFormatter": "biomejs.biome"
-  },
-  "[typescriptreact]": {
+  "[typescript][json][javascript][jsonc][typescriptreact]": {
```
Comment (idkgene) [inline .vscode/settings.json:11]: This formatter settings are being overriden by L3.

Comment (haydenbleasel) [reply .vscode/settings.json:11]: That's what I thought as well, but when I started using Biome, VS Code kept reverting to Prettier as the default formatter.

Comment (idkgene) [reply .vscode/settings.json:11]: > That's what I thought as well, but when I started using Biome, VS Code kept reverting to Prettier as the default formatter.

Seems I've found the issue here. The core problem stemmed from Biome not being installed globally on the system. This absence led to Biome's inoperability, causing VS Code to fall back to Prettier for all formatting operations. After installing Biome shows in this menu

image

Here's a sample code snipppet to test Output - Prettier and Output - Biome respectively

```
function  testFunction ( a , b ) {
    if(a > b) {
return  a;
} else {
    return  b;
}
}
```

So, the problem was resolved by installing Biome globally. However, the "format on save" functionality is not working on my machine with this

I will play with this behavior a little bit and I'll be back with updates on this soon

**Comment by idkgene** [comment]
This should work now, can you try it on your machine, @haydenbleasel ? Also what's your thoughts on the provided extensions.json file is it enough for now?

**Comment by haydenbleasel** [comment]
Made some changes to the extensions so they're more required and less opinionated, but looks good 👍

**Pull request merged**